Applications/Spire/Include/Spire/StyleParser/DataTypes/DataType.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef SPIRE_DATA_TYPE_HPP
+#define SPIRE_DATA_TYPE_HPP
+#include <string>
+
+namespace Spire {
+
+  /** Represents the abstract type. */
+  class DataType {
+    public:
+      virtual ~DataType() = default;
+
+      /** Returns the name. */
+      const std::string& get_name() const;
+
+      bool operator ==(const DataType& rhs) const;
+
+      bool operator !=(const DataType& rhs) const;
```

Comment:
I think this can be defaulted or even removed.

Reply:
Yes, it can be removed since "The inequality operator is automatically generated by the compiler if operator== is defined."

Referring to https://en.cppreference.com/w/cpp/language/operators